scope.go

```diff
@@ -0,0 +1,92 @@
+package rcmgr
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/libp2p/go-buffer-pool"
+	"github.com/libp2p/go-libp2p-core/network"
+)
+
+type ResourceScope struct {
+	sync.Mutex
+
+	limit   Limit
+	memory  int64
+	buffers map[interface{}][]byte
```

Comment:
Why do we need to store the buffers here?

Reply:
so that we can auto reclaim with Done in txn scopes

Reply:
added `releaseBuffers` utility method to use in transactional scopes.